extensions/amp-timeago/0.2/timeago.js

```diff
@@ -0,0 +1,60 @@
+/**
+ * Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {createElement, useRef, useState} from '../../../src/preact';
+import {timeago} from '../../../third_party/timeagojs/timeago';
+import {useInViewEffect} from '../../../src/preact/use-in-view';
+import {useResourcesNotify} from '../../../src/preact/utils';
+
+/**
+ * @param {!JsonObject} props
+ * @return {Preact.Renderable}
+ */
+export function Timeago(props) {
+  const {0: timestamp, 1: setTimestamp} = useState(
+    getFuzzyTimestampValue(props)
+  );
+
+  const ref = useRef(null);
+  useInViewEffect(
+    ref.current,
```

Comment:
As noted above, passing node here this way creates an issue. On the first pass, `ref.current == null`. It will not be assigned the value to the actual node until the the rendering is complete. Assigning this ref value itself will not cause re-render. So, unless something else will force an extra rerender, the `useInViewEffect` will be called only once and with the value `null`.